Installing a package as a static library for Linux/mac target should not require patchelf to be installed

**Describe the bug**

The current regex for excluding files does not exclude static libraries (.a files). Static libraries cannot have an rpath, and they will be rejected by patchelf since they don't start with the ELF header. Therefore, static libraries should be excluded.

This issue arose while cross-building a custom package from Windows to Linux, using a custom triplet to set the GCC compiler path. To avoid requiring patchelf on the developer's system, it's necessary to exclude static libraries. Since only a static binary is being built, patchelf isn't needed.

Currently, `z_vcpkg_fixup_rpath_in_dir` is always called when building for Linux, which mandates the use of patchelf even for static builds.

**Environment**
- OS: Windows 11
- Compiler: GCC

**Expected behavior**
A clear and concise description of what you expected to happen.

I would expect that .a files will not require patchelf to run, since static libraries don't have a rpath

**Failure logs**
-(please attached failure logs)

```
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:166 (message):
  Could not find patchelf.  Please install it via your package manager.
Call Stack (most recent call first):
  scripts/cmake/z_vcpkg_fixup_rpath.cmake:96 (vcpkg_find_acquire_program)
  scripts/ports.cmake:196 (z_vcpkg_fixup_rpath_in_dir)
```

**Additional context**
Add any other context about the problem here.

I confirmed locally that changing

`list(FILTER elf_files EXCLUDE REGEX "\\\.(cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")`

to

`list(FILTER elf_files EXCLUDE REGEX "\\\.(a|cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")`

resolves my issue.

--- a/scripts/cmake/z_vcpkg_fixup_rpath.cmake
+++ b/scripts/cmake/z_vcpkg_fixup_rpath.cmake
@@ -85,7 +85,7 @@ function(z_vcpkg_fixup_rpath_in_dir)
         endif()
 
         file(GLOB_RECURSE elf_files LIST_DIRECTORIES FALSE "${folder}/*")
-        list(FILTER elf_files EXCLUDE REGEX "\\\.(cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")
+        list(FILTER elf_files EXCLUDE REGEX "\\\.(a|cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")
         list(FILTER elf_files EXCLUDE REGEX "/(copyright|LICENSE|METADATA)$")
 
         foreach(elf_file IN LISTS elf_files)

